pages/sign-up.tsx

```diff
@@ -0,0 +1,289 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { VisibilityOff, Visibility } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  FormControl,
+  FormHelperText,
+  Grid,
+  IconButton,
+  InputAdornment,
+  InputLabel,
+  OutlinedInput,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import { grey } from "@mui/material/colors";
+import Link from "next/link";
+import React from "react";
+import { useForm, Controller } from "react-hook-form";
+import { ISignUp } from "../types/form.type";
+import client from "../apollo-client";
+import { SIGNUP_MUTATION } from "../graphql/sign-up";
+import { toast } from "react-hot-toast";
+import { useAppDispatch } from "../redux/redux-hook";
+import { ISignUpResponse } from "../types/graphql.respose";
+import { storeUser } from "../redux/slices/userSilce";
+import { useRouter } from "next/router";
+
+export default function SignUp() {
+  const theme = useTheme();
+  const router = useRouter();
+
+  const dispatch = useAppDispatch();
+  const { control, handleSubmit, watch } = useForm<ISignUp>({
+    defaultValues: {
+      email: "",
+      fullName: "",
+      password: "",
+      rePassword: "",
+    },
+  });
+
+  const [showPassword, setShowPassword] = React.useState(false);
+  const [showRepassword, setShowRepassword] = React.useState(false);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
```

Comment:
```suggestion
  const handleClickShowPassword = () => setShowPassword(!show);
```
có thể chạy đúng với mong đợi của em ko?